core/src/main/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactory.java

```diff
@@ -156,11 +157,22 @@ public boolean hasDescendingOrder() {
         return base.hasDescendingOrder();
     }
 
-    private static void filter(int workerId, PageAddressCacheRecord record, PageFrameReduceTask task) {
+    private static void filter(
+            int workerId,
+            PageAddressCacheRecord record,
+            PageFrameReduceTask task,
+            @Nullable PageFrameSequence<?> stealingFrameSequence
+    ) {
         final DirectLongList rows = task.getRows();
         final long frameRowCount = task.getFrameRowCount();
         final FilterAtom atom = task.getFrameSequence(FilterAtom.class).getAtom();
-        final Function filter = atom.getFilter(workerId);
+        final Function filter;
+        if (stealingFrameSequence != null && stealingFrameSequence != task.getFrameSequence(FilterAtom.class)) {
```

Comment:
What would happen in the following scenario?

A user has configured a dedicated PG Wire pool. Then they start sending queries with a filter both over PG Wire and REST API. Those queries get executed on threads with 0 id on both PG Wire and shared worker pool and then both of them start work stealing. They would end with accessing the same filter concurrently.

Potentially, to protect from that we'd need to make sure that all worker ids are unique. In this case, we could use a single array of filter functions.